### PR TITLE
Fix TS/RA2 tileset importer and apply correctness fixes to TS temperate.

### PR DESCRIPTION
**Pull request by pchote**
Extends #7707 to TS, and updates the tileset importer to use the correct terrain types and yaml keys.

### CODE REVIEW

**Comment by reaperrr** [comment]
The import command works, spotted no regressions ingame either.

I did run into one weird situation with the tileset I generated, I'd suddenly get a 'cannot connect to server' error when trying to access the skirmish menu. That was likely a problem on my end, though, I used a custom mix file generated by XCC Mixer, maybe something went wrong there.

**Comment by Mailaender** [comment]
I also noted the "cannot connect to server" (very seldom, < 1:100) when starting up #4496 and it is never reproducible, maybe a very unlikely network socket conflict. It is probably unrelated.

File: OpenRA.Mods.TS/UtilityCommands/LegacyTilesetImporter.cs [hunk 1]
```diff
@@ -95,7 +100,7 @@ public void Run(ModData modData, string[] args)
 								var terrainType = s.ReadUInt8();
 								var rampType = s.ReadUInt8();
 
-								if (!terrainTypes.ContainsKey(terrainType))
+								if (terrainType < 0 || terrainType > 15)
```
Comment (RoosterDragon) [inline OpenRA.Mods.TS/UtilityCommands/LegacyTilesetImporter.cs]: Maybe compare to ```terrainTypes.Length``` to avoid the magic constant. Also you appear to be testing if a byte can be negative.

**Comment by reaperrr** [comment]
> I also noted the "cannot connect to server" (very seldom, < 1:100) when starting up #4496 and it is never reproducible, maybe a very unlikely network socket conflict. It is probably unrelated.

I couldn't reproduce it this morning, either, and I haven't made any changes in the mean time, so I think you're right that it's unrelated.
My +1 still stands (once RoosterDragon's comment has been adressed in some way).

File: OpenRA.Mods.TS/UtilityCommands/LegacyTilesetImporter.cs (hunk 1 above)
Comment (pchote) [inline OpenRA.Mods.TS/UtilityCommands/LegacyTilesetImporter.cs]: Good catch. Fixed.
